test/commands/multi.spec.js

```diff
@@ -198,18 +198,55 @@ describe("The 'multi' method", function () {
                         });
                 });
 
-                it('reports multiple exceptions when they occur', function (done) {
-                    helper.serverVersionAtLeast.call(this, client, [2, 6, 5]);
+                it('reports EXECABORT exceptions when they occur (while queueing)', function (done) {
+                    client.multi().config("bar").set("foo").exec(function (err, reply) {
+                        assert.equal(err.code, "EXECABORT");
+                        assert.equal(reply, undefined, "The reply should have been discarded");
+                        assert(err.message.match(/^EXECABORT/), "Error message should begin with EXECABORT");
+                        assert.equal(err.errors.length, 1, "err.errors should have 1 items");
+                        assert(/^ERR/.test(err.errors[0].message), "Actuall error message should begin with ERR");
+                        return done();
+                    });
+                });
 
-                    client.multi().set("foo").exec(function (err, reply) {
-                        assert(Array.isArray(err), "err should be an array");
-                        assert.equal(2, err.length, "err should have 2 items");
-                        assert(err[0].message.match(/^ERR/), "First error message should begin with ERR");
-                        assert(err[1].message.match(/^EXECABORT/), "First error message should begin with EXECABORT");
+                it('reports multiple exceptions when they occur (while EXEC is running)', function (done) {
+                    client.multi().config("bar").debug("foo").exec(function (err, reply) {
+                        assert.equal(err.code, "MULTIERROR");
+                        assert(err.message.match(/^EXEC/), "Error message should begin with EXEC");
+                        assert.equal(err.errors.length, 2, "err.errors should have 2 items");
+                        assert(/^ERR/.test(err.errors[0].message), "Actuall error message should begin with ERR");
                         return done();
                     });
                 });
 
+                it('reports multiple exceptions when they occur (while EXEC is running) and calls cb', function (done) {
+                    var multi = client.multi();
+                    multi.config("bar", helper.isError());
+                    multi.set('foo', 'bar', helper.isString('OK'));
+                    multi.debug("foo").exec(function (err, reply) {
+                        assert.equal(err.code, "MULTIERROR");
+                        assert(err.message.match(/^EXEC/), "Error message should begin with EXEC");
+                        assert.equal(err.errors.length, 2, "err.errors should have 2 items");
+                        assert(/^ERR/.test(err.errors[0].message), "Actuall error message should begin with ERR");
+                        client.get('foo', helper.isString('bar', done));
```

Comment:
As you can see here, the value is indeed persistent even though an error occured.